source/system/Item.h

```diff
@@ -430,6 +373,207 @@ struct Book : public Item
+		case IT_CONSUMABLE:
+			prop = consumable;
+			break;
+		case IT_BOOK:
+			prop = book;
+			break;
+		case IT_OTHER:
+			prop = other_item;
+			break;
+		}
+		return static_cast<Prop*>(prop);
+	}
+	/**
+	 * Sets some property of this Item
+	 * Sets property's Item to this instance
+	 */
+	template<typename Type>
+	void SetProperty(ItemProp* p)
+	{
+		switch (Type::type)
+		{
+		case IT_WEAPON:
+			weapon = (Weapon*)p;
+			break;
+		case IT_BOW:
+			bow = (Bow*)p;
+			break;
+		case IT_SHIELD:
+			shield = (Shield*)p;
+			break;
+		case IT_ARMOR:
+			armor = (Armor*)p;
+			break;
+		case IT_AMULET:
+			amulet = (Amulet*)p;
+			break;
+		case IT_RING:
+			ring = (Ring*)p;
+			break;
+		case IT_CONSUMABLE:
+			consumable = (Consumable*)p;
+			break;
+		case IT_BOOK:
+			book = (Book*)p;
+			break;
+		case IT_OTHER:
+			other_item = (OtherItem*)p;
+			break;
+		default:
+			assert(0);
+			break;
+		}	
+		p->item = this;
+	}
+
+	/**
+	 * Retrieve various Item's properties.
+	 */
+	/*@{*/
+	Weapon& ToWeapon() { return *GetProperty<Weapon>(); }
+	Bow& ToBow() { return *GetProperty<Bow>(); }
+	Shield& ToShield() { return *GetProperty<Shield>(); }
+	Armor& ToArmor() { return *GetProperty<Armor>(); }
+	Amulet& ToAmulet() { return *GetProperty<Amulet>(); }
+	Ring& ToRing() { return *GetProperty<Ring>(); }
+	Consumable& ToConsumable() { return *GetProperty<Consumable>(); }
+	OtherItem& ToOther() { return *GetProperty<OtherItem>(); }
+	Book& ToBook() { return *GetProperty<Book>(); }
+
+	const Weapon& ToWeapon() const { return *weapon; }
+	const Bow& ToBow() const { return *bow; }
+	const Shield& ToShield() const { return *shield; }
+	const Armor& ToArmor() const { return *armor; }
+	const Amulet& ToAmulet() const { return *amulet; }
+	const Ring& ToRing() const { return *ring; }
+	const Consumable& ToConsumable() const { return *consumable; }
+	const OtherItem& ToOther() const { return *other_item; }
+	const Book& ToBook() const { return *book; }
+	/*@}*/
+
+	bool IsStackable() const { return Any(type, IT_CONSUMABLE, IT_GOLD, IT_BOOK) || (type == IT_OTHER && !IsSet(flags, ITEM_QUEST)); }
+	bool CanBeGenerated() const { return !IsSet(flags, ITEM_NOT_RANDOM); }
+	bool IsWearable() const { return Any(type, IT_WEAPON, IT_BOW, IT_SHIELD, IT_ARMOR, IT_AMULET, IT_RING); }
+	bool IsQuest() const { return IsSet(flags, ITEM_QUEST); }
+	bool IsQuest(int quest_id) const { return IsQuest() && this->quest_id == quest_id; }
+
+	const string& GetName() const { return name; }
+	float GetWeight() const { return float(weight) / 10; }
+	float GetWeightValue() const { return float(value) / weight; }
+	float GetEffectPower(EffectId effect) const;
+
+	void CreateCopy(Item& item) const;
+	Item* CreateCopy() const;
+	Item* QuestCopy(Quest* quest, const string& name);
+	void Rename(cstring name);
+	void RenameS(const string& name) { Rename(name.c_str()); }
+
+	string id, name, desc;
+	int weight, value, ai_value, flags, quest_id;
+	vector<ItemEffect> effects;
+	ITEM_TYPE type;
```

Comment:
This is because there are many places throughout the code where type is used for checking. I do not know the code well enough to safely remove it. It can treated like Item's *"Primary Type"*. I agree that in future `type` field should be removed.